### PR TITLE
[YUNIKORN-2231] Show node list when hovering mouse over the node utitutilization bar chart

### PR DESCRIPTION
**Pull request by chenyulin0719**
### What is this PR for?
- Enable Tooltip for bar-chart/area-chart/donut-chart.
- Show node list in tooltips’s footer. (No more than 15 nodes)
- Adjust ‘Nodes Resource Utilization’ chart width size (was 35%, is 50%)
- Use same color for bars in ‘Nodes Resource Utilization’ (Using multi-clor for the same resource is meaningless.)
- Expose domaint resource type in subtitle of ‘Nodes Resource Utilization’ bar cahrt.
- Fix wrong variable name in bar-chart. (was ‘donutChartData’, is ‘barChartData’)
- Refactor SchedulerService.fetchNodeUtilization() - move data transformation logic to NodeUtilization.toNodeUtilizationChartData() and add test case

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
NA

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2231

### How should this be tested?
Run 'make test' and all test cases should pass.  
Run 'yarn start' + 'yarn start:srv' in different terminal and check localhost:4200/

### Screenshots (if appropriate)
YUNIKORN-2231 before vs after

### Questions:
NA

### CODE REVIEW

File: src/app/models/node-utilization.model.ts [hunk 1]
```diff
@@ -25,4 +28,42 @@ export class NodeUtilization {
...
+
+  // transform NodeUtilization to NodeUtilizationChartData for NodeUtilization bar chart
+  toNodeUtilizationChartData(){
+    const MAX_NODES_IN_DESCRIPTION = 15;
+    const backgroundColor = CHART_COLORS[0];
+    let type = this.type;
+    let utilization = this.utilization;
+    // prepare data
+    let chartDataItems = new Array<ChartDataItem>();
+    let index = 0;
```
Comment (wusamzong) [inline src/app/models/node-utilization.model.ts:40]: This variable is declared but never used.

File: src/app/models/node-utilization.model.ts [hunk 2]
```diff
@@ -25,4 +28,42 @@ export class NodeUtilization {
       nodeNames: null | string[];
     }[],
   ) {}
+
+  // transform NodeUtilization to NodeUtilizationChartData for NodeUtilization bar chart
+  toNodeUtilizationChartData(){
+    const MAX_NODES_IN_DESCRIPTION = 15;
+    const backgroundColor = CHART_COLORS[0];
```
Comment (wusamzong) [inline src/app/models/node-utilization.model.ts:35]: I have a small question. Is it intended for backgroundColor to always be CHART_COLORS[0]? In the original implementation, the bar's background color would change. However, in this implementation, the background color remains constant, so there seems to be no need to declare an array CHART_COLORS with multiple colors.

File: src/app/utils/constants.ts [hunk 3]
```diff
@@ -20,3 +20,4 @@ export const DEFAULT_PARTITION_VALUE = '';
 export const DEFAULT_PROTOCOL = 'http:';
 export const NOT_AVAILABLE = 'n/a';
 export const PARTITION_DEFAULT = 'default';
+export const CHART_COLORS = ['#4285f4', '#db4437', '#f4b400', '#0f9d58', '#ff6d00', '#3949ab', '#facc54', '#26bbf0', '#cc6164', '#60cea5'];
```
Comment (wusamzong) [inline src/app/utils/constants.ts:23]: same question above.

**Review by wusamzong — CHANGES_REQUESTED** [review]
Most of this PR is great! But there are only a few minor problems.

File: src/app/models/node-utilization.model.ts (hunk 1 above)
Comment (chenyulin0719) [reply src/app/models/node-utilization.model.ts]: Just removed, sorry for missing that.

File: src/app/models/node-utilization.model.ts (hunk 2 above)
Comment (chenyulin0719) [reply src/app/models/node-utilization.model.ts]: Hi @wusamzong,
You're right, no need to declare multiple color here.  Just changed CHART_COLORS(Array) to CHART_COLOR(constant).

Thanks for the review.

File: src/app/utils/constants.ts (hunk 3 above)
Comment (chenyulin0719) [reply src/app/utils/constants.ts]: Updated.

**Review by wusamzong — APPROVED** [review]
LGTM

**Pull request closed without merging**